### PR TITLE
Initial renaming hipscat -> hats

### PR DESCRIPTION
**Pull request by delucchi-cmu**
## Change Description

This is a bulk renaming of HiPSCat -> HATS in most code. The most notable exception is the `hipscat_id`, which will require more finesse.

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -10,4 +10,5 @@ sphinx-autoapi
 sphinx-copybutton
 sphinx-book-theme
 sphinx-design
-git+https://github.com/astronomy-commons/hipscat.git@main
+git+https://github.com/astronomy-commons/hipscat.git@hats
```
Comment (gitosaurus) [inline docs/requirements.txt:13]: Noting that the name of the repo itself still has "hipscat" in it.  I assume that's a change we're obliged to make in a separate PR, but I'm just wanting to note it.

File: docs/tutorials/performance.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Performance
 LSDB is a high-performance package built to support the analysis of large-scale astronomical datasets.
 One of the performance goals of LSDB is to add as little overhead over the input-output operations as possible.
 We achieve this aim for catalog cross-matching, spatial and data filtering operations by using
-the `HiPSCat <https://github.com/astronomy-commons/hipscat>`_ data format,
+the `HATS <https://github.com/astronomy-commons/hats>`_ data format,
```
Comment (gitosaurus) [inline docs/tutorials/performance.rst:7]: Unless I miss my guess, this is assuming that the repo name has changed, doesn't it?

File: tests/data/small_sky_order1/provenance_info.json [hunk 3]
```diff
@@ -44,9 +44,11 @@
                 "chunksize": 500000,
                 "header": "infer",
                 "schema_file": null,
-                "separator": ",",
                 "column_names": null,
-                "type_map": {}
+                "type_map": null,
```
Comment (gitosaurus) [inline tests/data/small_sky_order1/provenance_info.json:48]: Just verifying that the change from empty to `null` is intentional.

File: tests/data/small_sky_to_o1source/catalog_info.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "primary_column": "id",
     "primary_column_association": "object_id",
     "join_catalog": "small_sky_order1_source",
-    "join_column": "source_id",
+    "join_column": "object_id",
```
Comment (gitosaurus) [inline tests/data/small_sky_to_o1source/catalog_info.json:9]: HATS change, or an additional refactoring during renaming?

File: tests/lsdb/catalog/test_crossmatch.py [hunk 5]
```diff
@@ -124,6 +124,7 @@ def test_right_margin_missing(algo, small_sky_catalog, small_sky_xmatch_catalog)
             small_sky_catalog.crossmatch(small_sky_xmatch_catalog, algorithm=algo, require_right_margin=True)
 
 
+@pytest.mark.skip(reason="HIPSCAT2HATS")
```
Comment (gitosaurus) [inline tests/lsdb/catalog/test_crossmatch.py:127]: 😎 I didn't know about this decorator. Nice!

File: tests/lsdb/core/test_kdtree_validation.py [hunk 6]
```diff
@@ -31,7 +31,7 @@ def test_bounded_kdtree_radius_invalid(small_sky_catalog, small_sky_order1_sourc
 def test_kdtree_left_columns(small_sky_catalog, small_sky_order1_source_with_margin):
     original_df = small_sky_catalog._ddf
     small_sky_catalog._ddf = original_df.reset_index()
-    with pytest.raises(ValueError, match="index of left table must be _hipscat_index"):
+    with pytest.raises(ValueError, match="index of left table must be _healpix_29"):
```
Comment (gitosaurus) [inline tests/lsdb/core/test_kdtree_validation.py:34]: I've seen this in a few other places.  Out of curiosity, why not `_hats_index`?

File: pyproject.toml [hunk 7]
```diff
@@ -21,7 +21,7 @@ dependencies = [
     # dask diagnostics is required to spin up the dashboard for profiling.
     "dask[complete]",
     "deprecated",
-    "hipscat>=0.3.8",
+    "hipscat>=0.3.9",
```
Comment (gitosaurus) [inline pyproject.toml:24]: Again, presumably the library/repository changes are upcoming.

Comment (delucchi-cmu) [reply pyproject.toml:24]: Yes - the repo/pypi release is still named 'hipscat', and I wanted the CI to still work until we can fully rename everything.

File: tests/lsdb/core/test_kdtree_validation.py (hunk 6 above)
Comment (delucchi-cmu) [reply tests/lsdb/core/test_kdtree_validation.py:34]: We want to be clearer that this is not a unique index to be used for primary keys.

File: tests/lsdb/catalog/test_crossmatch.py (hunk 5 above)
Comment (delucchi-cmu) [reply tests/lsdb/catalog/test_crossmatch.py:127]: Nice, right? Makes it pretty easy to find the cases we know are broken for now, and have CI still work for the ones that SHOULD work.

File: tests/data/small_sky_to_o1source/catalog_info.json (hunk 4 above)
Comment (delucchi-cmu) [reply tests/data/small_sky_to_o1source/catalog_info.json:9]: Hm. Not sure. The old value looks wrong, and the new value looks right, since this is the column to use for merging to the joining table.

File: tests/data/small_sky_order1/provenance_info.json (hunk 3 above)
Comment (delucchi-cmu) [reply tests/data/small_sky_order1/provenance_info.json:48]: Meh. This whole file will be going away sometime this week anyway. =D

File: tests/data/small_sky/provenance_info.json [hunk 8]
```diff
@@ -5,48 +5,50 @@
...
-            "overwrite": true,
-            "dask_tmp": "",
+            "tmp_dir": "/tmp/tmp0mmgvkr2",
+            "dask_tmp": null,
             "dask_n_workers": 1,
             "dask_threads_per_worker": 1,
-            "catalog_path": "./small_sky",
-            "tmp_path": "/tmp/user/11115/tmphywoxno9/small_sky/intermediate",
+            "catalog_path": "small_sky",
+            "tmp_path": "/tmp/tmp0mmgvkr2/small_sky/intermediate",
```
[the remark this answers is not in the document]
Comment (delucchi-cmu) [reply tests/data/small_sky/provenance_info.json:22]: Yes - part of the reason these provenance files will be removed / optional in this monster-migration.

https://github.com/astronomy-commons/hipscat/issues/351

File: src/lsdb/core/crossmatch/abstract_crossmatch_algorithm.py [hunk 9]
```diff
@@ -7,9 +7,9 @@
 import numpy as np
 import numpy.typing as npt
 import pandas as pd
-from hipscat.catalog.catalog_info import CatalogInfo
-from hipscat.catalog.margin_cache import MarginCacheCatalogInfo
-from hipscat.pixel_math.hipscat_id import HIPSCAT_ID_COLUMN
+from hats.catalog.catalog_info import CatalogInfo
+from hats.catalog.margin_cache import MarginCacheCatalogInfo
+from hats.pixel_math.hipscat_id import SPATIAL_INDEX_COLUMN
```
[the remark this answers is not in the document]
Comment (delucchi-cmu) [reply src/lsdb/core/crossmatch/abstract_crossmatch_algorithm.py:12]: Yes - it was indexing the data based on spatial characteristics, with a small counter to address duplicate locations. It was never a true "ID".

File: docs/tutorials/performance.rst (hunk 2 above)
Comment (delucchi-cmu) [reply docs/tutorials/performance.rst:7]: Yeah - it is. This will be a 404 for another week. We'll need to do a full scan of our documentation once everything has moved to update all references. But sphinx doesn't check that links aren't dead, so CI will still be ok.

File: docs/requirements.txt (hunk 1 above)
Comment (delucchi-cmu) [reply docs/requirements.txt:13]: Yes - we need to wait until the repo has actually been renamed, which should be relatively soon.

**Review by delucchi-cmu — COMMENTED** [review]
Thanks for looking in detail!

**Review by gitosaurus — APPROVED** [review]
